types/mapcmd/map_it_test.go

```diff
@@ -0,0 +1,448 @@
+		require.NoError(t, cmd.ExecuteContext(ctx))
+		v, err := m.Get(ctx, "test")
+		require.NoError(t, err)
+		require.Equal(t, "data", v)
+		v, err = m.Get(ctx, "json value")
+		require.NoError(t, err)
+		require.IsType(t, serialization.JSON{}, v)
+		vjson := v.(serialization.JSON)
+		require.JSONEq(t, `{"test" : "data"}`, string(vjson))
+	})
+}
+
+func TestMapPutAll(t *testing.T) {
+	it.MapTesterWithConfigAndMapName(t, func(t *testing.T, c *hazelcast.Config, m *hazelcast.Map, mn string) {
+		mapNameArg := fmt.Sprintf(` --name %s`, mn)
+		tcs := []struct {
+			name        string
+			args        string
+			entries     []types.Entry
+			value       interface{}
+			errContains string
+		}{
+			{
+				name: "valid put-all key(string), value(string)",
+				args: "-k k1 -k k2 -v v1 -v v2" + mapNameArg,
+				entries: []types.Entry{
+					{
+						Key:   "k1",
+						Value: "v1",
+					},
+					{
+						Key:   "k2",
+						Value: "v2",
+					},
+				},
+			},
+			{
+				name: "valid put-all key(json), value(json)",
+				args: `--key-type json --value-type json -k {"some":"key"} -v {"some":"value"} -k "test" -v "string"` + mapNameArg,
+				entries: []types.Entry{
+					{
+						Key:   serialization.JSON(`{"some":"key"}`),
+						Value: serialization.JSON(`{"some":"value"}`),
+					},
+					{
+						Key:   serialization.JSON(`test`),
+						Value: serialization.JSON(`string`),
+					},
+				},
+			},
+			{
+				name:        "invalid put-all, missing key",
+				args:        "--value v1" + mapNameArg,
+				errContains: `keys and values do not match`,
+			},
+			{
+				name:        "invalid put-all, missing map name",
+				args:        "--key k1 --value v1",
+				errContains: `"name" not set`,
+			},
+		}
+		for _, tc := range tcs {
+			t.Run(tc.name, func(t *testing.T) {
+				cmd := NewPutAll(c)
+				var stdout, stderr bytes.Buffer
+				cmd.SetOut(&stdout)
+				cmd.SetErr(&stderr)
+				args := strings.Split("./something "+tc.args, " ")
+				// no way other than this for put-all. validateJsonEntryFlag need the actual args to decide the order. Cobra don't pass the actual args
+				//cmd.SetArgs(args)
+				os.Args = args
+				ctx := context.Background()
+				_, err := cmd.ExecuteContextC(ctx)
+				if tc.errContains != "" {
+					require.Contains(t, err.Error(), tc.errContains)
+					return
+				}
+				require.NoError(t, err)
+				// cmd should not print anything
+				require.Empty(t, stdout.String())
+				require.Empty(t, stderr.String())
+				for _, e := range tc.entries {
+					value, err := m.Get(ctx, e.Key)
+					require.NoError(t, err)
+					fmt.Println(value)
+					require.Equal(t, e.Value, value)
+				}
+			})
+		}
+	})
+}
+
+func TestMapGet(t *testing.T) {
+	it.MapTesterWithConfigAndMapName(t, func(t *testing.T, c *hazelcast.Config, m *hazelcast.Map, mn string) {
+		mapNameArg := fmt.Sprintf(` --name %s`, mn)
+		tcs := []struct {
+			name        string
+			args        string
+			key         interface{}
+			value       interface{}
+			sout        string
```

Comment:
What's `sout`?

Reply:
renamed to `output` https://github.com/hazelcast/hazelcast-commandline-client/pull/114/commits/33e5d0a48d545a710b929043f671ab0808485bcf